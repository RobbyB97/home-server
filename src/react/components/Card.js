import React from 'react'
import {Link} from 'react-router-dom'

const Card = ({image, title, subtext, link}) => (
    <div>
        <div className="card__wrap">
            <a className="card__link" href={link}>
                <div className="card">
                    <img src={image}/>
                    <div className="card__text">
                        <h2>{title}</h2>
                        <p>{subtext}</p>
                    </div>
                </div>
            </a>
        </div>
    </div>
)

export default Card