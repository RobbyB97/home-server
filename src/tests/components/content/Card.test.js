import React from 'react'
import {shallow} from 'enzyme'

import Card from '../../../react/components/content/Card'


let wrapper

beforeEach(() => {
    wrapper = shallow(
        <Card 
            image="Image!"
            title="Title!"
            subtext="Subtext!"
            link="Link!"
        />
    )
})


test('Render Card', () => {
    expect(wrapper).toMatchSnapshot()
})