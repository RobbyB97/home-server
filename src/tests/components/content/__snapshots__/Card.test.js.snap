// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Card 1`] = `
<div>
  <div
    className="card"
  >
    <a
      className="card__link"
      href="Link!"
    >
      <div
        className="card__content"
      >
        <img
          src="Image!"
        />
        <div
          className="card__text"
        >
          <h2>
            Title!
          </h2>
          <p>
            Subtext!
          </p>
        </div>
      </div>
    </a>
  </div>
</div>
`;
